packages/ai-constructs/src/conversation/runtime/user_agent_provider.test.ts

```diff
@@ -0,0 +1,65 @@
+import { describe, it } from 'node:test';
+import assert from 'node:assert';
+import * as fs from 'node:fs';
+import path from 'path';
+import { UserAgentProvider } from './user_agent_provider';
+import { ConversationTurnEvent } from './types';
+
+void describe('User Agent provider', () => {
+  // Read package json from disk (i.e., in a different way than actual implementation does).
+  const packageVersion = JSON.parse(
+    fs.readFileSync(
+      path.resolve(__dirname, '..', '..', '..', 'package.json'),
+      'utf-8'
+    )
+  ).version;
```

Comment:
Curious why we are testing with getting package version in a different way than the implementation?

Reply:
To avoid tautology. I.e. using feeding data for assertion using copy of implementation would be that.

Otherwise, the assertion would need to be - there's some version in UA, it matches semver regex. (actual value is not constant).